### PR TITLE
Deprecate `ClusterSingleton.Init()` and add missing singleton feature to `ClusterSingletonSettings`

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

* Deprecate `ClusterSingleton.Init()` method and all of its supporting classes
* Add missing singleton detection feature to `ClusterSingletonSettings`

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@Arkatufus looks like API approvals are missing

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs [hunk 1]
```diff
@@ -22,6 +23,18 @@ public class ClusterSingleton : IExtension
     {
         private readonly ActorSystem _system;
         private readonly Lazy<Cluster> _cluster;
+        
+        /// <summary>
+        /// Returns default HOCON configuration for the cluster singleton.
+        /// </summary>
+        /// <returns>TBD</returns>
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs]: Remove the `TBD`

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs [hunk 2]
```diff
@@ -22,6 +23,18 @@ public class ClusterSingleton : IExtension
...
+        public static Config DefaultConfig()
+        {
+            return ConfigurationFactory.FromResource<ClusterSingleton>(
+                "Akka.Cluster.Tools.Singleton.reference.conf");
+        }
+        
+        // Cache for singleton proxies, remove in v1.6
         private readonly ConcurrentDictionary<string, IActorRef> _proxies = new();
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs:37]: Yeah I agree we don't really need a cache for this - that's a little too clever for its own good.

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs [hunk 3]
```diff
@@ -40,6 +54,10 @@ public ClusterSingleton(ExtendedActorSystem system)
         /// <para>If there already is a proxy running for the given `singletonName` on this node, an <see cref="IActorRef"/> to that is returned.</para>
         /// </summary>
         /// <returns>A proxy actor that can be used to communicate with the singleton in the cluster</returns>
+        [Obsolete("This convenience method is deprecated and will be removed in v1.6, " +
+                  "please use ClusterSingletonManager.Props and ClusterSingletonProxy.Props directly instead. " +
+                  "See https://getakka.net/community/whats-new/akkadotnet-v1.5-upgrade-advisories.html#upgrading-to-akkanet-v1532. " +
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs:58]: LGTM

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Need to add API approvals and do one other minor bit of cleanup

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs (hunk 1 above)
Comment (Arkatufus) [reply src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingleton.cs]: Done

**Comment by Arkatufus** [comment]
Updated the API Approval list

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**